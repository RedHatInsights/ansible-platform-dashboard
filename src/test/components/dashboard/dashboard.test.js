import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { MemoryRouter } from 'react-router-dom';
import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import analyticsReducer, { analyticsInitialState } from '../../../redux/reducers/analytics-reducer';
import hubReducer, { hubInitialState } from '../../../redux/reducers/hub-reducer';
import Dashboard from '../../../components/dashboard/dashboard';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { applyReducerHash, ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import UserContext from '../../../user-context';

const ComponentWrapper = ({ store, initialEntries = [ '/ansible-dashboard' ], children }) => (
  <Provider store={ store } >
    <MemoryRouter initialEntries={ initialEntries }>
      <IntlProvider locale="en">
        <UserContext.Provider
          value={ { permissions: []} }
        >
          { children }
        </UserContext.Provider>
      </IntlProvider>
    </MemoryRouter>
  </Provider>
);

describe('<Dashboard />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  beforeEach(() => {
    global.insights = {
      chrome: {
        identifyApp: jest.fn(),
        init: jest.fn(),
        auth: {
          getUser: () => Promise.resolve(true)
        }
      }
    };
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = {
      i18nReducer: {
        formatMessage: ({ defaultMessage }) => defaultMessage
      },
      analyticsReducer: { ...analyticsInitialState, isLoading: false },
      hubReducer: { ...hubInitialState, isLoading: false }
    };
  });

  it('should render correctly', async ()=> {
    const store = mockStore(initialState);
    const registry = new ReducerRegistry({}, [ thunk, promiseMiddleware ]);
    registry.register({ analyticsReducer: applyReducerHash(analyticsReducer, analyticsInitialState),
      hubReducer: applyReducerHash(hubReducer, hubInitialState) });

    const { asFragment } = render(<ComponentWrapper store={ store }><Dashboard { ...initialProps } /></ComponentWrapper>);

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly in loading state', () => {
    const store = mockStore(initialState);
    const { asFragment } = render(<ComponentWrapper store={ store }><Dashboard { ...initialProps } isLoading={ true } /></ComponentWrapper>);
    expect(asFragment()).toMatchSnapshot();
  });
});
