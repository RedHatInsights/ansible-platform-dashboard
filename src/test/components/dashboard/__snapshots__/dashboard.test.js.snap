// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render correctly 1`] = `
<DocumentFragment>
  <section
    class="pf-c-page__main-section pf-m-limit-width ans-c-marketing-banner ans-m-with-graphic  ans-m-graphic-right ans-m-light-1000 pf-m-light-1000 ans-m-full-bleed"
    style="--ins-c-marketing-banner--graphic--width-on-md: 200px; --ins-c-marketing-banner--graphic--width-on-xl: 400px;"
  >
    <div
      class="pf-c-page__main-body"
    >
      <div
        class="pf-l-grid"
      >
        <div
          class="pf-l-grid__item"
        >
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Overview
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <article
          class="pf-c-card ans-c-card-dashboard"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h3
              class="pf-c-title pf-m-lg"
              data-ouia-component-id="OUIA-Generated-Title-2"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Automation Analytics
            </h3>
          </div>
          <div
            class="pf-c-card__body pf-u-pb-sm"
          >
            <section
              class=""
              style="background-color: white; min-height: 100%;"
            >
               
              <div
                class="pf-l-bullseye"
              >
                <svg
                  aria-label="Contents"
                  aria-valuetext="Loading..."
                  class="pf-c-spinner pf-m-xl"
                  role="progressbar"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="pf-c-spinner__path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="45"
                  />
                </svg>
              </div>
               
            </section>
          </div>
        </article>
      </div>
      <div
        class="pf-l-stack__item"
      >
        <article
          class="pf-c-card ans-c-card-dashboard"
          data-ouia-component-id="OUIA-Generated-Card-2"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h3
              class="pf-c-title pf-m-lg"
              data-ouia-component-id="OUIA-Generated-Title-3"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Automation Hub
            </h3>
          </div>
          <div
            class="pf-c-card__body"
          >
            <section
              class=""
              style="background-color: white; min-height: 100%;"
            >
               
              <div
                class="pf-l-bullseye"
              >
                <svg
                  aria-label="Contents"
                  aria-valuetext="Loading..."
                  class="pf-c-spinner pf-m-xl"
                  role="progressbar"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="pf-c-spinner__path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="45"
                  />
                </svg>
              </div>
               
            </section>
          </div>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<Dashboard /> should render correctly in loading state 1`] = `
<DocumentFragment>
  <section
    class="pf-c-page__main-section pf-m-limit-width ans-c-marketing-banner ans-m-with-graphic  ans-m-graphic-right ans-m-light-1000 pf-m-light-1000 ans-m-full-bleed"
    style="--ins-c-marketing-banner--graphic--width-on-md: 200px; --ins-c-marketing-banner--graphic--width-on-xl: 400px;"
  >
    <div
      class="pf-c-page__main-body"
    >
      <div
        class="pf-l-grid"
      >
        <div
          class="pf-l-grid__item"
        >
          <div
            class="pf-l-flex pf-m-column"
          >
            <div
              class=""
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-4"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Overview
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <article
          class="pf-c-card ans-c-card-dashboard"
          data-ouia-component-id="OUIA-Generated-Card-3"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h3
              class="pf-c-title pf-m-lg"
              data-ouia-component-id="OUIA-Generated-Title-5"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Automation Analytics
            </h3>
          </div>
          <div
            class="pf-c-card__body pf-u-pb-sm"
          >
            <section
              class=""
              style="background-color: white; min-height: 100%;"
            >
               
              <div
                class="pf-l-bullseye"
              >
                <svg
                  aria-label="Contents"
                  aria-valuetext="Loading..."
                  class="pf-c-spinner pf-m-xl"
                  role="progressbar"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="pf-c-spinner__path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="45"
                  />
                </svg>
              </div>
               
            </section>
          </div>
        </article>
      </div>
      <div
        class="pf-l-stack__item"
      >
        <article
          class="pf-c-card ans-c-card-dashboard"
          data-ouia-component-id="OUIA-Generated-Card-4"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h3
              class="pf-c-title pf-m-lg"
              data-ouia-component-id="OUIA-Generated-Title-6"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Automation Hub
            </h3>
          </div>
          <div
            class="pf-c-card__body"
          >
            <section
              class=""
              style="background-color: white; min-height: 100%;"
            >
               
              <div
                class="pf-l-bullseye"
              >
                <svg
                  aria-label="Contents"
                  aria-valuetext="Loading..."
                  class="pf-c-spinner pf-m-xl"
                  role="progressbar"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="pf-c-spinner__path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="45"
                  />
                </svg>
              </div>
               
            </section>
          </div>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
